crates/hyperqueue/src/common/serverdir.rs

```diff
@@ -82,20 +82,28 @@ fn resolve_active_directory(path: &Path) -> PathBuf {
 }
 
 fn serde_serialize_key<S: Serializer>(
-    key: &Arc<SecretKey>,
+    key: &Option<Arc<SecretKey>>,
     serializer: S,
 ) -> Result<S::Ok, S::Error> {
-    let str = serialize_key(key);
-    serializer.serialize_str(&str)
+    if let Some(key) = key {
+        let str = serialize_key(key);
+        serializer.serialize_str(&str)
```

Comment:
Shouldn't this be `serialize_some`? :thinking:

Reply:
I think it does not matter, but you are right serializing an Option should be probably more elegant.

Reply:
If it works without it, then it's fine.